hazelcast/src/main/java/com/hazelcast/transaction/impl/xa/XAResourceImpl.java

```diff
@@ -107,7 +107,7 @@ public void start(Xid xid, int flags) throws XAException {
 
     private TransactionContext createTransactionContext(Xid xid) {
         XAService xaService = getService();
-        TransactionContext context = xaService.newXATransactionContext(xid, timeoutInSeconds.get(), false);
+        TransactionContext context = xaService.newXATransactionContext(xid, null, timeoutInSeconds.get(), false);
```

Comment:
I haven't check the full scope of this change, but intuitively the `null` here seems suspicious. Isn't it moving the problem (=missing UUID) to some other case?